dockers/docker-dhcp-server/cli/config/plugins/dhcp_server.py

```diff
@@ -186,6 +186,72 @@ def dhcp_server_ipv4_disable(db, dhcp_interface):
         ctx.fail("Failed to disable, dhcp interface {} does not exist".format(dhcp_interface))
 
 
+@dhcp_server_ipv4.group(cls=clicommon.AliasedGroup, name="range")
+def dhcp_server_ipv4_range():
+    pass
+
+
+def count_ipv4(start, end):
+    ip1 = int(ipaddress.IPv4Address(start))
+    ip2 = int(ipaddress.IPv4Address(end))
+    return ip2 - ip1 + 1
+
+
+@dhcp_server_ipv4_range.command(name="add")
+@click.argument("range_name", required=True)
+@click.argument("ip_start", required=True)
+@click.argument("ip_end", required=False)
+@clicommon.pass_db
+def dhcp_server_ipv4_range_add(db, range_name, ip_start, ip_end):
+    ctx = click.get_current_context()
+    if not ip_end:
+        ip_end = ip_start
+    if not validate_str_type("ipv4-address", ip_start) or not validate_str_type("ipv4-address", ip_end):
+        ctx.fail("ip_start or ip_end is not valid ipv4 address")
+    if count_ipv4(ip_start, ip_end) < 1:
+        ctx.fail("range value is illegal")
+    dbconn = db.db
+    key = "DHCP_SERVER_IPV4_RANGE|" + range_name
+    if dbconn.exists("CONFIG_DB", key):
+        ctx.fail("Range {} already exist".format(range_name))
+    else:
+        db.hmset("CONFIG_DB", key, {"range": ip_start + "," + ip_end})
+
+
+@dhcp_server_ipv4_range.command(name="update")
+@click.argument("range_name", required=True)
+@click.argument("ip_start", required=True)
+@click.argument("ip_end", required=False)
+@clicommon.pass_db
+def dhcp_server_ipv4_range_update(db, range_name, ip_start, ip_end):
+    ctx = click.get_current_context()
+    if not ip_end:
+        ip_end = ip_start
+    if not validate_str_type("ipv4-address", ip_start) or not validate_str_type("ipv4-address", ip_end):
+        ctx.fail("ip_start or ip_end is not valid ipv4 address")
+    if count_ipv4(ip_start, ip_end) < 1:
+        ctx.fail("range value is illegal")
+    dbconn = db.db
+    key = "DHCP_SERVER_IPV4_RANGE|" + range_name
+    if dbconn.exists("CONFIG_DB", key):
+        db.set("CONFIG_DB", key, "range", ip_start + "," + ip_end)
+    else:
+        ctx.fail("Range {} does not exist, cannot update".format(range_name))
+
+
+@dhcp_server_ipv4_range.command(name="del")
+@click.argument("range_name", required=True)
+@clicommon.pass_db
+def dhcp_sever_ipv4_range_del(db, range_name):
+    ctx = click.get_current_context()
+    dbconn = db.db
+    key = "DHCP_SERVER_IPV4_RANGE|" + range_name
+    if dbconn.exists("CONFIG_DB", key):
```

Comment:
How about add checking to make sure range is not used in DHCP_SERVER_IPV4 table and add a not required  argument '--force' to bypass the check?

Reply:
#Closed